src/UI/Settings.js

```diff
@@ -1062,7 +1087,11 @@ function Settings(props) {
         </Form>
         <Form onSubmit={handleSubmit}>
           <H3>Change text color</H3>
-          <Input placeholder="Hex or string" ref={textColorInput} />
+          <Input
+            defaultValue={theme.text_color}
```

Comment:
You should not rely on the theme, instead rely on the useTheme() object that has access to all the theme attributes like this:
useTheme().primary_color
You have multiple examples in this file to see how it's used.